### ISSUE TITLE
Add Resolved VanityUrl as request attribute

### ISSUE THREAD

**Issue by wezell** [dotCMS/core#19498]
When a vanityUrl is hit, it makes sense to add it to the request.attributes so it can be used or reported on further down the request without having to re-resolve it.

**Comment by nollymar** [comment]
PR: https://github.com/dotCMS/core/pull/19499

**Comment by dsilvam** [comment]
Note to QA: The variable should be available under `$request.getAttribute("VANITY_URL_OBJECT")`

**Comment by dsilvam** [comment]
Passed Internal QA: upon requesting the property mentioned above, we get the following output:
```
CachedVanityUrl{pattern=/costa-rica, vanityUrlId='048a19dc-2719-40e0-84b8-58175a90ec20', url='/costa-rica', siteId='SYSTEM_HOST', languageId=1, forwardTo='/destinations/costa-rica', response=200, order=0}
```

**Comment by bryanboza** [comment]
Fixedm tested on release-20.11 // Postgres // FF

**Comment by jcastro-dotcms** [comment]
The official code changes have been included in the dotCMS 5.3.8.4 LTS release only. The 5.2.8.4 LTS release is several commits behind, which made the code back-port not possible.
